src/pkg/artifact/dao/dao.go

```diff
@@ -180,7 +180,27 @@ func (d *dao) Update(ctx context.Context, artifact *Artifact, props ...string) e
 	if err != nil {
 		return err
 	}
-	n, err := ormer.Update(artifact, props...)
+
+	var newProps []string
+	for _, prop := range props {
+		// if prop is PullTime and before the PullTime in db,  just ignore the
+		// props to skip update.
+		if prop == "PullTime" {
+			old, err := d.Get(ctx, artifact.ID)
+			if err != nil {
+				return err
+			}
+
+			if artifact.PullTime.Before(old.PullTime) {
+				artifact.PullTime = old.PullTime
+				continue
+			}
+		}
+
+		newProps = append(newProps, prop)
+	}
+
+	n, err := ormer.Update(artifact, newProps...)
```

Comment:
I prefer to add a new method to only update the pull time in the DAO.

Reply:
good point